ibc-core/ics02-client/types/src/events.rs

```diff
@@ -219,6 +220,88 @@ impl From<CreateClient> for abci::Event {
         }
     }
 }
+// impl TryFrom<abci::Event> for CreateClient {
+//     type Error = ...;
+
+//     fn try_from(value: abci::Event) -> Result<Self, Self::Error> {
+//         (value.kind == CREATE_CLIENT_EVENT)
+//             .then(|| { ... })
+//             .ok_or_else(|| { ... })
+//     }
+// }
+impl TryFrom<abci::Event> for CreateClient {
+    type Error = ClientError;
+
+    fn try_from(value: abci::Event) -> Result<Self, Self::Error> {
+        if value.kind != CREATE_CLIENT_EVENT {
+            return Err(ClientError::Other {
+                description: "Error in parsing CreateClient event".to_string(),
+            });
+        }
+
+        let mut client_id = None;
+        let mut client_type = None;
+        let mut consensus_height = None;
+
+        for attribute in value.attributes.iter() {
+            let key = attribute.key_str().map_err(|_| ClientError::Other {
+                description: "Invalid attribute key".to_string(),
+            })?;
+
+            match key {
+                CLIENT_ID_ATTRIBUTE_KEY => {
+                    client_id = Some(parse_attribute_value(
+                        attribute,
+                        |value| ClientId::from_str(value).map_err(|_| ClientError::Other {
+                            description: "Invalid client ID attribute value".to_string(),
+                        }),
+                    )?);
+                }
+                CLIENT_TYPE_ATTRIBUTE_KEY => {
+                    client_type = Some(parse_attribute_value(
+                        attribute,
+                        |value| ClientType::from_str(value).map_err(|_| ClientError::Other {
+                            description: "Invalid client type attribute value".to_string(),
+                        }),
+                    )?);
+                }
+                CONSENSUS_HEIGHT_ATTRIBUTE_KEY => {
+                    consensus_height = Some(parse_attribute_value(
+                        attribute,
+                        |value| Height::from_str(value).map_err(|_| ClientError::Other {
+                            description: "Invalid consensus height attribute value".to_string(),
+                        }),
+                    )?);
```

Comment:
This looks correct. Do you plan to implement all the `TryFrom` that I mention in my issue? Because, I would rather have `TryFrom<EventAttribute>` implemented for ClientId, ClientType, ConsensusHeight (and for other similar cases)

Reply:
yes I think this PR will consists of `TryFrom` implemenation for the client's set of events, also would be nice if you could break the original #1220 to some sub-issues as you mentioned.

Reply:
Thanks ! I will create those sub issues soon :raised_hands: